From: =?UTF-8?q?Rog=C3=A9rio=20Brito?= <rbrito@ime.usp.br>
Date: Thu, 7 Jun 2012 07:09:37 -0300
Subject: First try at removing some of mp4v2.

Forwarded: partially (only first hunk)
Signed-off-by: Rogério Brito <rbrito@ime.usp.br>

--- a/libhb/muxmp4.c
+++ b/libhb/muxmp4.c
@@ -1,3 +1,4 @@
+#if 0
 /* muxmp4.c
 
    Copyright (c) 2003-2012 HandBrake Team
@@ -1399,3 +1400,4 @@
     return m;
 }
 
+#endif /* 0*/
--- a/libhb/muxcommon.c
+++ b/libhb/muxcommon.c
@@ -460,8 +460,9 @@
         switch( job->mux )
         {
         case HB_MUX_MP4:
-            mux->m = hb_mux_mp4_init( job );
-            break;
+            hb_error( "MP4 muxer disabled, exiting" );
+            *job->die = 1;
+            return NULL;
         case HB_MUX_MKV:
             mux->m = hb_mux_mkv_init( job );
             break;
--- a/test/module.defs
+++ b/test/module.defs
@@ -13,7 +13,7 @@
 
 TEST.GCC.l = \
         a52 ass avcodec avformat avutil avresample dvdnav dvdread \
-        fontconfig freetype mkv mpeg2 mp3lame mp4v2 ogg \
+        fontconfig freetype mkv mpeg2 mp3lame ogg \
         samplerate swscale theoraenc theoradec vorbis vorbisenc x264 \
         bluray xml2 bz2 z
 
--- a/libhb/decmetadata.c
+++ b/libhb/decmetadata.c
@@ -7,10 +7,11 @@
    For full terms see the file COPYING file or visit http://www.gnu.org/licenses/gpl-2.0.html
  */
 
-#include <mp4v2/mp4v2.h>
-
 #include "common.h"
 
+#if 0
+#include <mp4v2/mp4v2.h>
+
 static int decmp4metadata( hb_title_t *title )
 {
     MP4FileHandle input_file;
@@ -172,6 +173,7 @@
     }
     return result;
 }
+#endif /* 0 */
 
 /*
  * decmetadata()
@@ -190,12 +192,5 @@
         return 0;
     }
 
-    /*
-     * Hacky way of figuring out if this is an MP4, in which case read the data using libmp4v2
-     */
-    if( title->container_name && strcmp(title->container_name, "mov,mp4,m4a,3gp,3g2,mj2") == 0 ) 
-    {
-        return decmp4metadata( title );
-    }
     return 0;
 }
--- a/gtk/src/hb-backend.c
+++ b/gtk/src/hb-backend.c
@@ -176,7 +176,6 @@
 static options_map_t d_container_opts[] =
 {
     {"MKV", "mkv", HB_MUX_MKV, "mkv"},
-    {"MP4", "mp4", HB_MUX_MP4, "mp4"},
 };
 combo_opts_t container_opts =
 {
--- a/gtk/src/presets.c
+++ b/gtk/src/presets.c
@@ -1394,7 +1394,7 @@
     str = ghb_settings_get_string(ud->settings, "destination_dir");
     ghb_ui_update(ud, "dest_dir", ghb_string_value(str));
 
-    gchar *file = g_strdup_printf ("new_video.mp4");
+    gchar *file = g_strdup_printf ("new_video.mkv");
     ghb_ui_update(ud, "dest_file", ghb_string_value(file));
     g_free(str);
     g_free(file);
--- a/gtk/src/internal_defaults.xml
+++ b/gtk/src/internal_defaults.xml
@@ -19,7 +19,7 @@
         <key>dest_dir</key>
         <string></string>
         <key>dest_file</key>
-        <string>new_video.mp4</string>
+        <string>new_video.mkv</string>
         <key>end_point</key>
         <integer>100</integer>
         <key>folder</key>
@@ -262,7 +262,7 @@
         <key>ChapterMarkers</key>
         <true />
         <key>FileFormat</key>
-        <string>mp4</string>
+        <string>mkv</string>
         <key>Folder</key>
         <false />
         <key>PictureLooseCrop</key>
--- a/gtk/configure.ac
+++ b/gtk/configure.ac
@@ -203,7 +203,7 @@
     ;;
 esac
 
-HB_LIBS="-lhb -la52 -lmkv -lavresample -lavformat -lavcodec -lavutil -ldvdnav -ldvdread -lmp3lame -lmpeg2 -lvorbis -lvorbisenc -logg -lsamplerate -lx264 -lmp4v2 -lswscale -ltheoraenc -ltheoradec -lz -lbz2 -lpthread -lbluray -lass -lfontconfig -lfreetype -lxml2"
+HB_LIBS="-lhb -la52 -lmkv -lavresample -lavformat -lavcodec -lavutil -ldvdnav -ldvdread -lmp3lame -lmpeg2 -lvorbis -lvorbisenc -logg -lsamplerate -lx264 -lswscale -ltheoraenc -ltheoradec -lz -lbz2 -lpthread -lbluray -lass -lfontconfig -lfreetype -lxml2"
 
 if test "x$use_fdk_aac" = "xyes" ; then
     HB_LIBS+=" -lfdk-aac"
