Author: Fabian Greffrath <fabian+debian@greffrath.com>
Subject: Consider CFLAGS, CPPFLAGS and LDFLAGS in libhb build.

Forwarded: yes
--- a/gtk/module.rules
+++ b/gtk/module.rules
@@ -17,7 +17,7 @@
 		$(GTK.CONFIGURE.extra) \
 		PKG_CONFIG_PATH=$(BUILD/)contrib/lib/pkgconfig \
 	    CFLAGS="$(call fn.ARGS,GTK.GCC,.g .O *D ?extra)" \
-	    LDFLAGS="$(call fn.ARGS,GTK.GCC,?strip .g .O) " \
+	    LDFLAGS="$(call fn.ARGS,GTK.GCC,?strip .g .O ?extra.exe) " \
 	    --prefix=$(PREFIX) \
 	    --with-hb=$(call fn.ABSOLUTE,$(BUILD/))
 	$(TOUCH.exe) $@
--- a/make/include/gcc.defs
+++ b/make/include/gcc.defs
@@ -67,15 +67,15 @@
 GCC.args.l         = -l$(1)
 GCC.args.end       = -Wl,--end-group
 
-GCC.args.extra         =
+GCC.args.extra         = $(CFLAGS) $(CPPFLAGS)
 GCC.args.extra.h_o     =
 GCC.args.extra.c_o     =
-GCC.args.extra.dylib   =
-GCC.args.extra.exe     =
+GCC.args.extra.dylib   = $(LDFLAGS)
+GCC.args.extra.exe     = $(LDFLAGS)
 GCC.args.extra.hpp_o   =
 GCC.args.extra.cpp_o   =
-GCC.args.extra.dylib++ =
-GCC.args.extra.exe++   =
+GCC.args.extra.dylib++ = $(LDFLAGS)
+GCC.args.extra.exe++   = $(LDFLAGS)
 
 ###############################################################################
 
