--- a/make/include/main.defs
+++ b/make/include/main.defs
@@ -8,8 +8,6 @@
 
 ###############################################################################
 
-MODULES += contrib/a52dec
-
 ifneq (,$(filter $(BUILD.system),cygwin mingw))
 ifneq ($(HAS.bz2),1)
     MODULES += contrib/bzip2
@@ -20,25 +18,8 @@
     MODULES += contrib/fribidi
 endif
 
-
-MODULES += contrib/faac
-MODULES += contrib/ffmpeg
-MODULES += contrib/fontconfig
-MODULES += contrib/freetype
-MODULES += contrib/lame
-MODULES += contrib/libass
-MODULES += contrib/libdca
 MODULES += contrib/libdvdread
 MODULES += contrib/libdvdnav
-MODULES += contrib/libbluray
-MODULES += contrib/libmkv
-MODULES += contrib/libogg
-MODULES += contrib/libsamplerate
-MODULES += contrib/libtheora
-MODULES += contrib/libvorbis
-MODULES += contrib/libxml2
-MODULES += contrib/mp4v2
 MODULES += contrib/mpeg2dec
 
 ifneq (,$(filter $(BUILD.system),mingw))
@@ -46,8 +29,6 @@
 endif
 endif
 
-MODULES += contrib/x264
-
 ifneq (,$(filter $(BUILD.system),cygwin mingw))
 ifneq ($(HAS.iconv),1)
     MODULES += contrib/libiconv
@@ -82,6 +63,11 @@
     ## build gtk when gtk+linux
     MODULES += gtk
 endif
+
+ifeq (1-kfreebsd,$(FEATURE.gtk)-$(BUILD.system))
+    ## build gtk when gtk+linux
+    MODULES += gtk
+endif
 
 ifeq (1,$(FEATURE.local_yasm))
     MODULES += contrib/yasm
--- a/contrib/ffmpeg/module.defs
+++ b/contrib/ffmpeg/module.defs
@@ -29,7 +29,7 @@
     --enable-encoder=snow \
     --enable-gpl \
     --enable-zlib \
-    --cc="$(FFMPEG.GCC.gcc)" \
+    --cc="ccache gcc" \
     --extra-cflags="$(call fn.ARGS,FFMPEG.GCC,*archs *sysroot *minver ?extra) -I$(call fn.ABSOLUTE,$(CONTRIB.build/)include)" \
     --extra-ldflags="$(call fn.ARGS,FFMPEG.GCC,*archs *sysroot *minver ?extra) -L$(call fn.ABSOLUTE,$(CONTRIB.build/)lib)"
 
--- a/test/module.defs
+++ b/test/module.defs
@@ -10,10 +10,11 @@
 TEST.exe = $(BUILD/)$(call TARGET.exe,$(HB.name)CLI)
 
 TEST.libs = $(LIBHB.a) $(foreach n, \
-        a52 ass avcodec avformat avutil dca dvdnav dvdread faac fontconfig freetype mkv mpeg2 mp3lame mp4v2 \
-        ogg samplerate swscale theora vorbis vorbisenc x264 xml2 bluray, \
+        dvdnav dvdread mpeg2, \
         $(CONTRIB.build/)lib/lib$(n).a )
 
+TEST.libs += -lass -lfaac -lmp3lame -lsamplerate -lvorbisenc -ltheoraenc -ltheoradec -lvorbis -lx264 -lmkv -lz -lbz2 -lpthread -ldl -lmp4v2 -lxml2 -ldca -la52 -lbluray -lavcodec -lavformat -lavutil -lswscale
+
 TEST.install.exe = $(DESTDIR)$(PREFIX/)bin/$(notdir $(TEST.exe))
 
 ifeq (1,$(PTHREADW32.enabled))
@@ -32,11 +33,6 @@
 else
 	TEST.GCC.l += z
 endif
-ifeq (1,$(FRIBIDI.enabled))
-    TEST.libs += $(CONTRIB.build/)lib/libfribidi.a
-else
-	TEST.GCC.l += fribidi
-endif
 
 ###############################################################################
 
--- a/gtk/src/Makefile.am
+++ b/gtk/src/Makefile.am
@@ -10,7 +10,7 @@
 HB_LIBS= \
 	-lhb -la52 -lmkv -lavformat -lavcodec -lavutil -ldca -ldvdnav -ldvdread \
 	-lfaac -lmp3lame -lmpeg2 -lvorbis -lvorbisenc -logg -lsamplerate \
-	-lx264 -lmp4v2 -lswscale -ltheora -lz \
+	-lx264 -lmp4v2 -lswscale -ltheoraenc -ltheoradec -lz -lxml2 \
 	-lbz2 -lpthread -lbluray -lass -lfontconfig -lfreetype
 endif
 
@@ -108,7 +108,7 @@
 	-mwindows -Wl,--export-dynamic -Wl,--exclude-libs,ALL
 else
 ghb_LDFLAGS = \
-	-Wl,--export-dynamic -Wl,--exclude-libs,ALL
+	-Wl,--export-dynamic -Wl,--exclude-libs,ALL -Wl,-z,defs -Wl,--as-needed
 endif
 
 ghb_LDADD = $(HB_LIBS) $(GHB_LIBS)
