Index: handbrake-0.9.3/test/Makefile
===================================================================
--- handbrake-0.9.3.orig/test/Makefile	2008-12-14 14:34:44.000000000 +0100
+++ handbrake-0.9.3/test/Makefile	2008-12-14 15:17:26.000000000 +0100
@@ -1,7 +1,6 @@
 SYSTEM = $(shell uname -s)
 
-LIBS = a52 mkv avformat avcodec avutil dca dvdread faac mp3lame mpeg2 vorbis \
-	   vorbisenc ogg samplerate x264 xvidcore mp4v2 swscale theora faad
+LIBS = a52 avformat avcodec avutil dca mpeg2 x264 mp4v2 swscale
 
 CYGLIBS = zlib bzip2
 
@@ -23,7 +22,7 @@
 	@CMD="$(CC) $(CFLAGS) -o parsecsv.o -c parsecsv.c"; $$CMD || \
 	( echo "Compile line for $@ was:"; echo $$CMD; false )
 	@echo "Link HandBrakeCLI"
-	@CMD="g++ $(CXXFLAGS) -o ../HandBrakeCLI test.o parsecsv.o $(LDFLAGS) -lz -lpthread"; $$CMD || \
+	@CMD="g++ $(CXXFLAGS) -o ../HandBrakeCLI test.o parsecsv.o $(LDFLAGS) -lmp3lame -ltheora -lfaad -lxvidcore -lvorbisenc -logg -lsamplerate -lvorbis -ldvdread -lfaac -lmkv -lz -lpthread"; $$CMD || \
 	( echo "Compile line for $@ was:"; echo $$CMD; false )
         @CMD="rm -rf ../plugins ; mkdir ../plugins ; cp ../contrib/lib/libquicktime/* ../plugins"; $$CMD
 
Index: handbrake-0.9.3/Jamfile
===================================================================
--- handbrake-0.9.3.orig/Jamfile	2008-12-14 14:34:44.000000000 +0100
+++ handbrake-0.9.3/Jamfile	2008-12-14 15:17:26.000000000 +0100
@@ -10,14 +10,9 @@
 HANDBRAKE_LIBS = libhb.a
           contrib/lib/liba52.a        contrib/lib/libavformat.a
           contrib/lib/libavcodec.a    contrib/lib/libavutil.a
-          contrib/lib/libdca.a        contrib/lib/libdvdread.a
-          contrib/lib/libmp4v2.a      contrib/lib/libfaac.a
-          contrib/lib/libmp3lame.a    contrib/lib/libmpeg2.a
-          contrib/lib/libvorbis.a     contrib/lib/libvorbisenc.a
-          contrib/lib/libogg.a        contrib/lib/libsamplerate.a
-          contrib/lib/libx264.a       contrib/lib/libxvidcore.a
-          contrib/lib/libmkv.a        contrib/lib/libswscale.a
-          contrib/lib/libtheora.a     contrib/lib/libfaad.a ;
+          contrib/lib/libdca.a        contrib/lib/libmp4v2.a
+          contrib/lib/libmpeg2.a      contrib/lib/libx264.a
+          contrib/lib/libswscale.a ;
 
 if $(OS) = UNKNOWN
 {
Index: handbrake-0.9.3/libhb/Makefile
===================================================================
--- handbrake-0.9.3.orig/libhb/Makefile	2008-12-14 14:34:44.000000000 +0100
+++ handbrake-0.9.3/libhb/Makefile	2008-12-14 15:17:26.000000000 +0100
@@ -31,15 +31,9 @@
 
 CONTRIBS = ../contrib/lib/liba52.a ../contrib/lib/libavformat.a \
 		   ../contrib/lib/libavcodec.a ../contrib/lib/libavutil.a \
-		   ../contrib/lib/libdca.a ../contrib/lib/libdvdread.a  \
-		   ../contrib/lib/libfaac.a ../contrib/lib/libmp3lame.a \
-		   ../contrib/lib/libmpeg2.a ../contrib/lib/libmpeg2convert.a \
-		   ../contrib/lib/libvorbis.a ../contrib/lib/libvorbisenc.a \
-		   ../contrib/lib/libvorbisfile.a ../contrib/lib/libogg.a \
-		   ../contrib/lib/libsamplerate.a ../contrib/lib/libx264.a \
-		   ../contrib/lib/libxvidcore.a  ../contrib/lib/libmp4v2.a \
-		   ../contrib/lib/libmkv.a ../contrib/lib/libswscale.a \
-		   ../contrib/lib/libtheora.a ../contrib/lib/libfaad.a
+		   ../contrib/lib/libdca.a ../contrib/lib/libmpeg2.a \
+		   ../contrib/lib/libmpeg2convert.a ../contrib/lib/libx264.a \
+		   ../contrib/lib/libmp4v2.a ../contrib/lib/libswscale.a
 		   
 ifeq ($(findstring CYGWIN_NT,$(SYSTEM)),CYGWIN_NT)
 	CONTRIBS += ../contrib/lib/libz.a ../contrib/lib/libbz2.a
Index: handbrake-0.9.3/contrib/Jamfile
===================================================================
--- handbrake-0.9.3.orig/contrib/Jamfile	2008-12-14 14:34:44.000000000 +0100
+++ handbrake-0.9.3/contrib/Jamfile	2008-12-14 15:17:26.000000000 +0100
@@ -57,23 +57,6 @@
 Wget   $(SUBDIR)/a52dec.tar.gz : $(SUBDIR)/version_a52dec.txt ;
 LibA52 $(SUBDIR)/lib/liba52.a  : $(SUBDIR)/a52dec.tar.gz ;
 
-# FAAD2
-rule LibFaad2
-{
-    Depends $(<) : $(>) ;
-    Depends lib  : $(<) ;
-}
-
-actions LibFaad2
-{
-    cd `dirname $(>)` && CONTRIB=`pwd` &&
-    rm -rf faad2 && (gzip -dc faad2.tar.gz | tar xf -) &&
-    cd faad2 && $(PATCH) -p1 < ../patch-faad2-libtool22.patch &&
-    ./bootstrap && ./configure --prefix=$CONTRIB --cache-file=$CONTRIB/config.cache --disable-shared && $(MAKE) && $(MAKE) install
-}
-Wget     $(SUBDIR)/faad2.tar.gz    : $(SUBDIR)/version_faad2.txt ;
-LibFaad2 $(SUBDIR)/lib/libfaad.a : $(SUBDIR)/faad2.tar.gz ;
-
 # libavcodec
 rule LibAvCodec
 {
@@ -173,90 +156,6 @@
 Wget   $(SUBDIR)/libdca.tar.gz : $(SUBDIR)/version_libdca.txt ;
 LibDCA $(SUBDIR)/lib/libdca.a  : $(SUBDIR)/libdca.tar.gz ;
 
-# libdvdread
-rule LibDvdRead
-{
-    Depends $(<) : $(>) ;
-    Depends lib  : $(<) ;
-    
-    if $(OS) = MACOSX
-    {
-        DVDREAD_PATCH = "$(PATCH) -p0 < ../patch-libdvdread-css-vlc-dylib.patch &&" ;
-    }
-	
-    if $(OS) = CYGWIN
-    {
-        DVDREAD_PATCH = "$(PATCH) -p1 < ../patch-libdvdread-cygwin.patch &&" ;
-    }
-    
-}
-if $(CSS) = 1 # Include CSS support in libdvdread if requested
-{
-    actions LibDvdRead
-    {
-        cd `dirname $(>)` && CONTRIB=`pwd` &&
-        rm -rf libdvdread && (gzip -dc libdvdread.tar.gz | tar xf - ) && 
-        cd libdvdread &&
-        ./configure --prefix=$CONTRIB --cache-file=$CONTRIB/config.cache --disable-shared --with-libdvdcss=$CONTRIB &&
-        $(MAKE) && $(MAKE) install &&
-        $(STRIP) $CONTRIB/lib/libdvdread.a
-    }
-}
-else {
-    # Cygwin/Windows doesn't use CSS as of now, so don't include it on the
-    # configure line.
-    # MacOSX uses a dylib for CSS, so it doesn't need including there either.
-    actions LibDvdRead
-    {
-        cd `dirname $(>)` && CONTRIB=`pwd` &&
-        rm -rf libdvdread && (gzip -dc libdvdread.tar.gz | tar xf - ) &&
-        cd libdvdread && $(DVDREAD_PATCH)
-        ./configure --prefix=$CONTRIB --cache-file=$CONTRIB/config.cache --disable-shared &&
-        $(MAKE) && $(MAKE) install &&
-        $(STRIP) $CONTRIB/lib/libdvdread.a
-    }
-}
-Wget       $(SUBDIR)/libdvdread.tar.gz : $(SUBDIR)/version_libdvdread.txt ;
-LibDvdRead $(SUBDIR)/lib/libdvdread.a  : $(SUBDIR)/libdvdread.tar.gz ;
-
-# libfaac
-rule LibFaac
-{
-    Depends $(<) : $(>) ;
-    Depends lib  : $(<) ;
-	if $(OS) = CYGWIN
-    {
-        CYGWIN_PATCH = "$(PATCH) -p1 < ../patch-faac-cygwin.patch &&" ;
-    }
-}
-actions LibFaac
-{
-    cd `dirname $(>)` && CONTRIB=`pwd` &&
-    rm -rf faac && (gzip -dc faac.tar.gz | tar xf - ) && cd faac && $(CYGWIN_PATCH) ./bootstrap &&
-    ./configure --prefix=$CONTRIB --cache-file=$CONTRIB/config.cache --disable-shared --enable-static --without-mp4v2 &&
-    $(MAKE) && $(MAKE) install &&
-    $(STRIP) $CONTRIB/lib/libfaac.a
-}
-Wget    $(SUBDIR)/faac.tar.gz   : $(SUBDIR)/version_faac.txt ;
-LibFaac $(SUBDIR)/lib/libfaac.a : $(SUBDIR)/faac.tar.gz ;
-
-# libmp3lame
-rule LibMp3Lame
-{
-    Depends $(<) : $(>) ;
-    Depends lib  : $(<) ;
-}
-actions LibMp3Lame
-{
-    cd `dirname $(>)` && CONTRIB=`pwd` &&
-    rm -rf lame && (gzip -dc lame.tar.gz | tar xf - ) && cd lame &&
-    ./configure --prefix=$CONTRIB --cache-file=$CONTRIB/config.cache --disable-shared &&
-    $(MAKE) && $(MAKE) install &&
-    $(STRIP) $CONTRIB/lib/libmp3lame.a
-}
-Wget       $(SUBDIR)/lame.tar.gz      : $(SUBDIR)/version_lame.txt ;
-LibMp3Lame $(SUBDIR)/lib/libmp3lame.a : $(SUBDIR)/lame.tar.gz ;
-
 # libmp4v2
 rule LibMp4v2
 {
@@ -275,23 +174,6 @@
 Wget     $(SUBDIR)/libmp4v2.tar.gz : $(SUBDIR)/version_libmp4v2.txt ;
 LibMp4v2 $(SUBDIR)/lib/libmp4v2.a : $(SUBDIR)/libmp4v2.tar.gz ;
 
-rule LibMkv
-{
-    Depends $(<) : $(>) ;
-    Depends lib  : $(<) ;
-}
-actions LibMkv
-{
-    cd `dirname $(>)` && CONTRIB=`pwd` &&
-    rm -rf libmkv && (gzip -dc libmkv.tar.gz | tar xf - ) && 
-    cd libmkv &&
-    ./configure --disable-shared --enable-static --prefix=$CONTRIB --cache-file=$CONTRIB/config.cache &&
-    $(MAKE) && $(MAKE) install &&
-    $(STRIP) $CONTRIB/lib/libmkv.a
-}
-Wget    $(SUBDIR)/libmkv.tar.gz : $(SUBDIR)/version_libmkv.txt ;
-LibMkv  $(SUBDIR)/lib/libmkv.a  : $(SUBDIR)/libmkv.tar.gz ;
-
 # libmpeg2
 rule LibMpeg2
 {
@@ -310,90 +192,6 @@
 Wget     $(SUBDIR)/mpeg2dec.tar.gz : $(SUBDIR)/version_mpeg2dec.txt ;
 LibMpeg2 $(SUBDIR)/lib/libmpeg2.a  : $(SUBDIR)/mpeg2dec.tar.gz ;
 
-# libogg
-rule LibOgg
-{
-    Depends $(<) : $(>) ;
-    Depends lib  : $(<) ;
-}
-actions LibOgg
-{
-    cd `dirname $(>)` && CONTRIB=`pwd` &&
-    rm -rf libogg && (gzip -dc libogg.tar.gz | tar xf - ) && 
-    cd libogg &&
-    ./configure --prefix=$CONTRIB --cache-file=$CONTRIB/config.cache --disable-shared &&
-    $(MAKE) && $(MAKE) install &&
-    $(STRIP) $CONTRIB/lib/libogg.a
-}
-Wget   $(SUBDIR)/libogg.tar.gz : $(SUBDIR)/version_libogg.txt ;
-LibOgg $(SUBDIR)/lib/libogg.a  : $(SUBDIR)/libogg.tar.gz ;
-
-# libsamplerate
-rule LibSampleRate
-{
-    Depends $(<) : $(>) ;
-    Depends lib  : $(<) ;
-}
-actions LibSampleRate
-{
-    cd `dirname $(>)` && CONTRIB=`pwd` &&
-    rm -rf libsamplerate && (gzip -dc libsamplerate.tar.gz | tar xf - ) && 
-    cd libsamplerate &&
-    ./configure --prefix=$CONTRIB --cache-file=$CONTRIB/config.cache --disable-shared &&
-    $(MAKE) && $(MAKE) install &&
-    $(STRIP) $CONTRIB/lib/libsamplerate.a
-}
-Wget          $(SUBDIR)/libsamplerate.tar.gz : $(SUBDIR)/version_libsamplerate.txt ;
-LibSampleRate $(SUBDIR)/lib/libsamplerate.a  : $(SUBDIR)/libsamplerate.tar.gz ;
-
-# libvorbis
-rule LibVorbis
-{
-    Depends $(<) : $(>) ;
-    Depends lib  : $(<) ;
-}
-actions LibVorbis
-{
-    cd `dirname $(>)` && CONTRIB=`pwd` &&
-    rm -rf libvorbis && (gzip -dc libvorbis.tar.gz | tar xf - ) &&
-    cd libvorbis &&
-    ./configure --prefix=$CONTRIB --with-ogg=$CONTRIB --cache-file=$CONTRIB/config.cache --disable-shared &&
-    $(MAKE) && $(MAKE) install &&
-    $(STRIP) $CONTRIB/lib/libvorbis.a
-}
-Wget      $(SUBDIR)/libvorbis.tar.gz : $(SUBDIR)/version_libvorbis.txt ;
-LibVorbis $(SUBDIR)/lib/libvorbis.a  : $(SUBDIR)/libvorbis.tar.gz ;
-
-rule LibVorbisEnc
-{
-    Depends $(<) : $(>) ;
-    Depends lib  : $(<) ;
-}
-actions LibVorbisEnc
-{
-    $(STRIP) $(<)
-}
-LibVorbisEnc $(SUBDIR)/lib/libvorbisenc.a : $(SUBDIR)/lib/libvorbis.a ;
-
-# libtheora
-rule LibTheora
-{
-    Depends $(<) : $(>) ;
-    Depends lib  : $(<) ;
-}
-actions LibTheora
-{
-    cd `dirname $(>)` && CONTRIB=`pwd` && rm -rf libtheora &&
-    (gzip -dc libtheora.tar.gz | tar xf - ) && cd libtheora &&
-    HAVE_PKG_CONFIG="no" && export HAVE_PKG_CONFIG &&
-    ./configure --prefix=$CONTRIB --with-ogg=$CONTRIB --with-vorbis=$CONTRIB --cache-file=$CONTRIB/config.cache --disable-shared \
-            --disable-oggtest --disable-vorbistest --disable-sdltest --disable-examples &&
-    $(MAKE) && $(MAKE) install &&
-    $(STRIP) $CONTRIB/lib/libtheora.a
-}
-Wget      $(SUBDIR)/libtheora.tar.gz : $(SUBDIR)/version_libtheora.txt ;
-LibTheora $(SUBDIR)/lib/libtheora.a  : $(SUBDIR)/libtheora.tar.gz ;
-
 # libx264
 rule LibX264
 {
@@ -421,36 +219,6 @@
 Wget    $(SUBDIR)/x264.tar.gz   : $(SUBDIR)/version_x264.txt ;
 LibX264 $(SUBDIR)/lib/libx264.a : $(SUBDIR)/x264.tar.gz ;
 
-# libxvidcore
-rule LibXvidCore
-{
-    LIBXVIDCORE_PATCH = "" ;
-    if $(OS) = MACOSX
-    {
-        LIBXVIDCORE_PATCH = "$(PATCH) -p1 < ../patch-xvidcore-macosx.patch && " ;
-    }
-    else if $(OS) = CYGWIN
-    {
-        LIBXVIDCORE_PATCH = "$(PATCH) -p1 < ../patch-xvidcore-cygwin.patch && " ;
-    }
-    LIBXVIDCORE_PATCH += "$(PATCH) -p1 < ../patch-xvidcore-nasm-2.00-configure.patch && " ;
-    LIBXVIDCORE_PATCH += "$(PATCH) -p1 < ../patch-xvidcore-fdct.patch && " ;
-    Depends $(<) : $(>) ;
-    Depends lib  : $(<) ;
-}
-actions LibXvidCore
-{
-    cd `dirname $(>)` && CONTRIB=`pwd` &&
-    rm -rf xvidcore && (gzip -dc xvidcore.tar.gz | tar xf - ) &&
-    cd xvidcore && $(LIBXVIDCORE_PATCH)
-    cd build/generic/ && ./configure --prefix=$CONTRIB --cache-file=$CONTRIB/config.cache && $(MAKE) libxvidcore.a &&
-    cp ./=build/libxvidcore.a $CONTRIB/lib/ &&
-    cp ../../src/xvid.h $CONTRIB/include/ &&
-    $(STRIP) $CONTRIB/lib/libxvidcore.a
-}
-Wget        $(SUBDIR)/xvidcore.tar.gz   : $(SUBDIR)/version_xvidcore.txt ;
-LibXvidCore $(SUBDIR)/lib/libxvidcore.a : $(SUBDIR)/xvidcore.tar.gz ;
-
 # zlib
 if $(OS) = CYGWIN
 {
Index: handbrake-0.9.3/Makefile
===================================================================
--- handbrake-0.9.3.orig/Makefile	2008-12-14 14:34:44.000000000 +0100
+++ handbrake-0.9.3/Makefile	2008-12-14 15:17:26.000000000 +0100
@@ -77,7 +77,6 @@
 	rm -f libhb/hbversion.h
 
 all:	contrib/.contrib libhb/hbversion.h HandBrakeCLI
-	(rm -rf HandBrake HandBrake*.tar.gz ; mkdir -p HandBrake/api HandBrake/doc; cp test/BUILDSHARED AUTHORS BUILD COPYING CREDITS NEWS THANKS TRANSLATIONS HandBrake/doc ;  cp -rp libhb/hb.h libhb/common.h libhb/ports.h HandBrake/api ; cp -rp HandBrakeCLI HandBrake ; tar zcvf HandBrake-$(HB_VERSION)_i386.tar.gz HandBrake ; rm -rf HandBrake )
 
 contrib/.contrib:
 	@$(MAKE) --no-print-directory -C contrib all
Index: handbrake-0.9.3/gtk/src/Makefile.am
===================================================================
--- handbrake-0.9.3.orig/gtk/src/Makefile.am	2008-12-14 14:34:44.000000000 +0100
+++ handbrake-0.9.3/gtk/src/Makefile.am	2008-12-14 15:17:40.000000000 +0100
@@ -7,25 +7,25 @@
 HB_LIBS=\
 	$(HB_DIR)/libhb/libhb.a \
 	$(HB_DIR)/contrib/lib/liba52.a \
-	$(HB_DIR)/contrib/lib/libmkv.a \
+	-lmkv \
 	$(HB_DIR)/contrib/lib/libavformat.a \
 	$(HB_DIR)/contrib/lib/libavcodec.a \
 	$(HB_DIR)/contrib/lib/libavutil.a \
 	$(HB_DIR)/contrib/lib/libdca.a \
-	$(HB_DIR)/contrib/lib/libdvdread.a \
-	$(HB_DIR)/contrib/lib/libfaac.a \
-	$(HB_DIR)/contrib/lib/libmp3lame.a \
+	-ldvdread \
+	-lfaac \
+	-lmp3lame \
 	$(HB_DIR)/contrib/lib/libmpeg2.a \
-	$(HB_DIR)/contrib/lib/libvorbis.a \
-	$(HB_DIR)/contrib/lib/libvorbisenc.a \
-	$(HB_DIR)/contrib/lib/libogg.a \
-	$(HB_DIR)/contrib/lib/libsamplerate.a \
+	-lvorbis \
+	-lvorbisenc \
+	-logg \
+	-lsamplerate \
 	$(HB_DIR)/contrib/lib/libx264.a \
-	$(HB_DIR)/contrib/lib/libxvidcore.a \
+	-lxvidcore \
 	$(HB_DIR)/contrib/lib/libmp4v2.a \
 	$(HB_DIR)/contrib/lib/libswscale.a \
-	$(HB_DIR)/contrib/lib/libtheora.a \
-	$(HB_DIR)/contrib/lib/libfaad.a \
+	-ltheora \
+	-lfaad \
 	-lz \
 	-lbz2 \
 	-lpthread
